packages/@n8n/task-runner/src/task-runner.ts

```diff
@@ -298,6 +300,20 @@ export abstract class TaskRunner extends EventEmitter {
 		}
 		task.cancelled = true;
 
+		for (const [requestId, request] of this.dataRequests.entries()) {
+			if (request.taskId === taskId) {
+				request.reject(new ApplicationError('Task cancelled'));
```

Comment:
We receive a cancellation reason in the message. We could include that here. Also could use some more fine grained error than generic `ApplicationError`

Reply:
[ffaba80](https://github.com/n8n-io/n8n/pull/12101/commits/ffaba80f79f637173bc14fed4c3346a5152f7235)